From d4a7cb9b73160cc6d7990d14bdce5e6a5cd141f1 Mon Sep 17 00:00:00 2001
From: Andreas Larsson <andreas@gaisler.com>
Date: Wed, 22 Dec 2021 14:39:54 +0100
Subject: [PATCH 26/32] phy/micrel: Add phy_id matching for KSZ9021GN

KSZ9021GN has phy_id 0x00221612, which has not been matched to KSZ9021
since commit 48d7d0ad9022 ("phy/micrel: change phy_id_mask for KSZ9021
and KS8001") made the mask too strict for that.

Signed-off-by: Andreas Larsson <andreas@gaisler.com>
---
 drivers/net/phy/micrel.c   | 4 ++--
 include/linux/micrel_phy.h | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/phy/micrel.c b/drivers/net/phy/micrel.c
index bbbe198f83e8..f879d64623a6 100644
--- a/drivers/net/phy/micrel.c
+++ b/drivers/net/phy/micrel.c
@@ -1292,7 +1292,7 @@ static struct phy_driver ksphy_driver[] = {
 	.resume		= genphy_resume,
 }, {
 	.phy_id		= PHY_ID_KSZ9021,
-	.phy_id_mask	= 0x000ffffe,
+	.phy_id_mask	= 0x000ffffc,
 	.name		= "Micrel KSZ9021 Gigabit PHY",
 	/* PHY_GBIT_FEATURES */
 	.driver_data	= &ksz9021_type,
@@ -1396,7 +1396,7 @@ MODULE_AUTHOR("David J. Choi");
 MODULE_LICENSE("GPL");
 
 static struct mdio_device_id __maybe_unused micrel_tbl[] = {
-	{ PHY_ID_KSZ9021, 0x000ffffe },
+	{ PHY_ID_KSZ9021, 0x000ffffc },
 	{ PHY_ID_KSZ9031, MICREL_PHY_ID_MASK },
 	{ PHY_ID_KSZ9131, MICREL_PHY_ID_MASK },
 	{ PHY_ID_KSZ8001, 0x00fffffc },
diff --git a/include/linux/micrel_phy.h b/include/linux/micrel_phy.h
index 416ee6dd2574..f2cc258bed1a 100644
--- a/include/linux/micrel_phy.h
+++ b/include/linux/micrel_phy.h
@@ -13,6 +13,7 @@
 #define PHY_ID_KSZ8873MLL	0x000e7237
 #define PHY_ID_KSZ9021		0x00221610
 #define PHY_ID_KSZ9021RLRN	0x00221611
+#define PHY_ID_KSC9021GN	0x00221612
 #define PHY_ID_KS8737		0x00221720
 #define PHY_ID_KSZ8021		0x00221555
 #define PHY_ID_KSZ8031		0x00221556
-- 
2.34.1

